### ISSUE TITLE
[Support]: Not detecting last TPU when using multiple cards

### ISSUE THREAD

**Issue by Flamingi** [blakeblackshear/frigate#2156]
### Describe the problem you are having

Whenever I try to run Frigate with multiple TPUs it doesn't find all cards. No matter if I specify 2, 3 or 4 cards it will always detect 1/2/3 TPUs correctly and on the last one I get a `Failed to load delegate from libedgetpu.so.1.0` error. It's also not always the same PCI device, it's always the last one that throws the error, i.e. without changing any configs from one run to another it may be any of the PCI devices, whichever is the last one it tries to detect.

### Version

0.9.4

### Frigate config file

```yaml
mqtt:
  host: 192.168.1.30
  port: 8883
  user: frigate
  password: password

detectors:
  coral1:
    type: edgetpu
    device: pci:0
  coral2:
    type: edgetpu
    device: pci:1
  coral3:
    type: edgetpu
    device: pci:2

record:
  enabled: True
  retain_days: 2
  events:
    retain:
      default: 100

cameras:
  <...>
```

### Relevant log output

```shell
[s6-init] making user provided files available at /var/run/s6/etc...exited 0.

[s6-init] ensuring user provided files have correct perms...exited 0.

[fix-attrs.d] applying ownership & permissions fixes...

[fix-attrs.d] done.

[cont-init.d] executing container initialization scripts...

[cont-init.d] done.

[services.d] starting services

[services.d] done.

[2021-10-31 18:05:30] frigate.app                    INFO    : Starting Frigate (0.9.4-26ae608)

Starting migrations

[2021-10-31 18:05:30] peewee_migrate                 INFO    : Starting migrations

There is nothing to migrate

[2021-10-31 18:05:30] peewee_migrate                 INFO    : There is nothing to migrate

[2021-10-31 18:05:30] frigate.mqtt                   INFO    : MQTT connected

[2021-10-31 18:05:30] detector.coral1                INFO    : Starting detection process: 241

[2021-10-31 18:05:30] detector.coral2                INFO    : Starting detection process: 243

[2021-10-31 18:05:30] detector.coral3                INFO    : Starting detection process: 246

[2021-10-31 18:05:30] frigate.app                    INFO    : Output process started: 250

[2021-10-31 18:05:30] frigate.app                    INFO    : Camera processor started for cam5: 260

[2021-10-31 18:05:30] frigate.app                    INFO    : Camera processor started for cam6: 270

[2021-10-31 18:05:30] frigate.edgetpu                INFO    : Attempting to load TPU as pci:2

[2021-10-31 18:05:30] ws4py                          INFO    : Using epoll

[2021-10-31 18:05:30] frigate.edgetpu                INFO    : TPU found

[2021-10-31 18:05:30] frigate.edgetpu                INFO    : Attempting to load TPU as pci:0

[2021-10-31 18:05:30] frigate.app                    INFO    : Camera processor started for cam7: 320

[2021-10-31 18:05:30] frigate.edgetpu                INFO    : TPU found

[2021-10-31 18:05:30] frigate.app                    INFO    : Camera processor started for cam2: 337

[2021-10-31 18:05:30] frigate.app                    INFO    : Capture process started for cam5: 339

[2021-10-31 18:05:30] frigate.app                    INFO    : Capture process started for cam6: 342

[2021-10-31 18:05:30] frigate.app                    INFO    : Capture process started for cam7: 346

[2021-10-31 18:05:30] frigate.app                    INFO    : Capture process started for cam2: 350

[2021-10-31 18:05:30] ws4py                          INFO    : Using epoll

[2021-10-31 18:05:30] frigate.edgetpu                INFO    : Attempting to load TPU as pci:1

Process detector:coral2:

[2021-10-31 18:05:43] frigate.edgetpu                ERROR   : No EdgeTPU was detected. If you do not have a Coral device yet, you must configure CPU detectors.

Traceback (most recent call last):

  File "/usr/lib/python3/dist-packages/tflite_runtime/interpreter.py", line 160, in load_delegate

    delegate = Delegate(library, options)

  File "/usr/lib/python3/dist-packages/tflite_runtime/interpreter.py", line 119, in __init__

    raise ValueError(capture.message)

ValueError


During handling of the above exception, another exception occurred:


Traceback (most recent call last):

  File "/usr/lib/python3.8/multiprocessing/process.py", line 315, in _bootstrap

    self.run()

  File "/usr/lib/python3.8/multiprocessing/process.py", line 108, in run

    self._target(*self._args, **self._kwargs)

  File "/opt/frigate/frigate/edgetpu.py", line 156, in run_detector

    object_detector = LocalObjectDetector(

  File "/opt/frigate/frigate/edgetpu.py", line 64, in __init__

    edge_tpu_delegate = load_delegate("libedgetpu.so.1.0", device_config)

  File "/usr/lib/python3/dist-packages/tflite_runtime/interpreter.py", line 162, in load_delegate

    raise ValueError('Failed to load delegate from {}\n{}'.format(

ValueError: Failed to load delegate from libedgetpu.so.1.0


[2021-10-31 18:05:50] frigate.watchdog               INFO    : Detection appears to have stopped. Exiting frigate...

[cont-finish.d] executing container finish scripts...

[cont-finish.d] done.

[s6-finish] waiting for services.

[2021-10-31 18:05:50] frigate.video                  INFO    : cam6: ffmpeg sent a broken frame. memoryview assignment: lvalue and rvalue have different structures

[2021-10-31 18:05:50] frigate.video                  INFO    : cam6: ffmpeg process is not running. exiting capture thread...

[2021-10-31 18:05:50] frigate.video                  INFO    : cam7: ffmpeg sent a broken frame. memoryview assignment: lvalue and rvalue have different structures

[2021-10-31 18:05:50] frigate.video                  INFO    : cam7: ffmpeg process is not running. exiting capture thread...

[2021-10-31 18:05:50] frigate.video                  INFO    : cam2: ffmpeg sent a broken frame. memoryview assignment: lvalue and rvalue have different structures

[2021-10-31 18:05:50] frigate.video                  INFO    : cam2: ffmpeg process is not running. exiting capture thread...

[2021-10-31 18:05:50] frigate.video                  INFO    : cam5: ffmpeg sent a broken frame. memoryview assignment: lvalue and rvalue have different structures

[2021-10-31 18:05:50] frigate.video                  INFO    : cam5: ffmpeg process is not running. exiting capture thread...

[s6-finish] sending all processes the TERM signal.

[s6-finish] sending all processes the KILL signal and exiting.
```

### FFprobe output from your camera

```shell
-
```

### Operating system

Debian

### Install method

Docker Compose

### Coral version

M.2

### Network connection

Wired

### Camera make and model

-

### Any other information that may be helpful

Docker-compose config:
```
#################### Frigate
  frigate:
    container_name: frigate
    privileged: true
    restart: unless-stopped
    environment:
      - TZ=Europe/Berlin
      - FRIGATE_RTSP_PASSWORD=password
    image: blakeblackshear/frigate:stable-amd64
    devices:
      - /bus/pci/devices/0000&#58;03&#58;00.0
      - /bus/pci/devices/0000&#58;09&#58;00.0
      - /bus/pci/devices/0000&#58;0b&#58;00.0
    volumes:
      - /home/flamingi/docker/frigate.yml:/config/config.yml:ro
      - /mnt/data/frigate/:/media/frigate
      - type: tmpfs
        target: /tmp/cache
        tmpfs:
          size: 1000000000
    ports:
      - 5000:5000
      - 1935:1935
    shm_size: '2gb'

```

**Comment by blakeblackshear** [comment]
Does it run reliably with 3 specified?

**Comment by Flamingi** [comment]
> Does it run reliably with 3 specified?

As long as I only use one in my Frigate config file it works.

**Comment by blakeblackshear** [comment]
I'm not sure why that would be the case. I know there are a few others with multiple Corals around here, so maybe someone can offer some advice.

**Comment by Flamingi** [comment]
> i think mine are running fine with that call... maybe worth a try?

I copied your config (minus the hwaccel) and unfortunately got the same error.

**Comment by Flamingi** [comment]
So I played around a bit more with the config and got it working using this docker-compose:
```
devices:
  - /dev/apex_0
  - /dev/apex_1
  - /dev/apex2
```
and this frigate config:
```
detectors:
  coral1:
    type: edgetpu
    device: pci:0
  coral2:
    type: edgetpu
    device: pci:1
  coral3:
    type: edgetpu
    device: pci:2
```

I also found out the following (which doesn't make any sense at all):
- In the testing scenarios I passed the cards in docker-compose like this: `/dev/apex_{i}:/dev/apex_{i}`
- If I only load one TPU in Frigate with `pci`, `pci:0` pr `pci:2` it works. however using `pci:1` or any other number (tested up to `pci:15`) does not work. It also doesn't matter which card(s) I have in my docker-compose. (I tried many, many combinations in order to find some causal link). Specifically, passing only 1 card in docker-compose and using `pci:2` does work.
- If I load two TPUs from `pci`/`pci:0` and `pci:2`, I can detect both cards. Using any other combination of ports will not work. Again, it doesn't matter which cards (and paths) I have in my docker-compose. 
- Only passing one TPU in docker-compose and specifying two (`pci:0` and `pci:2`) in Frigate works??
